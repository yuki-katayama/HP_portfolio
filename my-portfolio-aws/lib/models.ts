export interface SesForm {
    adminEmail: string,
    name: string,
    email: string,
    content: string,
}