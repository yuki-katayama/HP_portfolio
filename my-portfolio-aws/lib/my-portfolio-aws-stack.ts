import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {load} from 'ts-dotenv'
import { S3 } from './resources/s3';
import { Apigateway } from './resources/apigateway';
import { Cloudfront } from './resources/cloudfront';
import { Route53 } from './resources/route53';
import { Acm } from './resources/acm';
import { Lambda } from './resources/lambda';

const env = load({
  BacketName: String,
  ApigatewayName: String,
  DeployPathToBacket: String,
  DomainName: String,
  LambdaName: String,
});


export class MyPortfolioAwsStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    const s3 = new S3();
    const apigateway = new Apigateway();
    const cloudfront = new Cloudfront();
    const route53 = new Route53();
    const acm = new Acm();
    const lambda = new Lambda();

    super(scope, id, props);
      /* s3の作成 */
      s3.createResources(this, env.BacketName);
      /* cloudFrontのOAIを作成 */
      cloudfront.createOAI(this);
      /* cloudFrontからのアクセスをs3側で許可する */
      s3.enableAccessFromCloudFront(cloudfront.oai);
      /* s3へソースをデプロイ */
      s3.deploy(this, env.DeployPathToBacket);
      /* route53 作成 */
      route53.createResources(this, env.DomainName);
      /* 証明書発行 */
      acm.createResources(this, route53.publicHostedZone, env.DomainName);
      /* cloudFrontの作成 */
      cloudfront.createResources(this, env.DomainName, s3.s3Bucket, acm.certificatemanager);
      /* route53をcloudFrontへ繋ぐ */
      route53.registerArecord(this, cloudfront.distribution, env.DomainName);
      /* lambdaの作成 */
      lambda.createResources(this, env.LambdaName);
      /* apigatewayの作成 */
      apigateway.createResources(this, env.ApigatewayName);
      /* apigatewayにlambdaを繋げる */
  }
}
