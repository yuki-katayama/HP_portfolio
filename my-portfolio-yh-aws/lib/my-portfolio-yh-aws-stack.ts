import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';

import { Acm } from './resource/acm';
import { Route53 } from './resource/route53';
import { S3 } from './resource/s3';
import { CloudFront } from './resource/cloudfront';
import { Apigateway } from './resource/apigateway';

export class MyPortfolioYhAwsStack extends cdk.Stack {

  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const domainName = "tekitou.ga";
    const backetName = "my-port-folio-yh-tekitou-cdk";
    const deployPathToBacket = "./../deploy/";
    const apigatewayName = "sendMailTest"

    const route53 = new Route53();
    const s3 = new S3();
    const acm = new Acm();
    const cloudfront = new CloudFront();
    const apigateway = new Apigateway();

    /* s3の作成 */
    s3.createResources(this, backetName);
    /* cloudFrontのOAIを作成 */
    cloudfront.createOAI(this);
    /* cloudFrontからのアクセスをs3側で許可する */
    s3.enableAccessFromCloudFront(cloudfront.oai);
    /* s3へソースをデプロイ */
    s3.deploy(this, deployPathToBacket);
    /* route53 作成 */
    route53.createResources(this, domainName);
    /* 証明書発行 */
    acm.createResources(this, route53.publicHostedZone, domainName);
    /* cloudFrontの作成 */
    cloudfront.createResources(this, domainName, s3.s3Bucket, acm.certificatemanager);
    /* route53をcloudFrontへ繋ぐ */
    route53.registerArecord(this, cloudfront.distribution, domainName);
    apigateway.createResources(this, apigatewayName);
  }
}