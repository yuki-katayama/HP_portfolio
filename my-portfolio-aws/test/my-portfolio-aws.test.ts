import * as cdk from 'aws-cdk-lib';
import * as MyPortfolioAws from '../lib/my-portfolio-aws-stack';
import { Template } from 'aws-cdk-lib/assertions';
import {load} from 'ts-dotenv'
// import * as MyPortfolioAws from '../lib/my-portfolio-aws-stack';

const env = load({
	BacketName: String,
	ApigatewayName: String,
	DeployPathToBacket: String,
	DomainName: String,
  });

const app = new cdk.App();
const stack = new MyPortfolioAws.MyPortfolioAwsStack(app, 'MyTestStack');
// THEN
const template = Template.fromStack(stack);

test('Snapshot Is Correct', () => {
	expect(template).toMatchSnapshot();
})

test('S3 Bucket Created', () => {
	template.hasResourceProperties(
		'AWS::S3::Bucket', {
			"BucketEncryption": {
			  "ServerSideEncryptionConfiguration": [
				{
				  "ServerSideEncryptionByDefault": {
					"SSEAlgorithm": "AES256"
				  }
				}
			  ]
			},
			"PublicAccessBlockConfiguration": {
				"BlockPublicAcls": true,
				"IgnorePublicAcls": true,
			},
			"BucketName": env.BacketName,
			"WebsiteConfiguration": {
				"IndexDocument": "index.html",
			}
		})
	}
)

test('Route53 Bucket Created', () => {
	template.hasResourceProperties(
		"AWS::Route53::RecordSet", {
			"Name": env.DomainName + '.',
			"Type": "A",
		}
	)
})

test('Apigateway Created', () => {
	template.hasResourceProperties(
		'AWS::ApiGateway::RestApi', {
			"Name": env.ApigatewayName,
		}
	)
})