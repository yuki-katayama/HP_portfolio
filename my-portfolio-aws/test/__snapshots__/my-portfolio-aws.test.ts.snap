// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Is Correct 1`] = `
{
  "Mappings": {
    "AWSCloudFrontPartitionHostedZoneIdMap": {
      "aws": {
        "zoneId": "Z2FDTNDATAQYW2",
      },
      "aws-cn": {
        "zoneId": "Z3RFFRIM2A3IF5",
      },
    },
  },
  "Outputs": {
    "myportfoliotestEndpoint380B0B6E": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "myportfoliotest1BB247FD",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "myportfoliotestDeploymentStageprodA4566A1E",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "Alias325C5727": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "myDist9DB766F3",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": {
          "Ref": "PublicHostedZone49973947",
        },
        "Name": "tekitou.ga.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "CertificateManagerCertificateCertificateRequestorFunction9E036FFB": {
      "DependsOn": [
        "CertificateManagerCertificateCertificateRequestorFunctionServiceRoleDefaultPolicy6F05AD3D",
        "CertificateManagerCertificateCertificateRequestorFunctionServiceRole39CBE3CA",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "0affa35a9d750381b76be83ff9cedeb2e455e8903fcf4d7db4d1c1e2c9c7cd54.zip",
        },
        "Handler": "index.certificateRequestHandler",
        "Role": {
          "Fn::GetAtt": [
            "CertificateManagerCertificateCertificateRequestorFunctionServiceRole39CBE3CA",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CertificateManagerCertificateCertificateRequestorFunctionServiceRole39CBE3CA": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CertificateManagerCertificateCertificateRequestorFunctionServiceRoleDefaultPolicy6F05AD3D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "acm:RequestCertificate",
                "acm:DescribeCertificate",
                "acm:DeleteCertificate",
                "acm:AddTagsToCertificate",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "route53:GetChange",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "route53:changeResourceRecordSets",
              "Condition": {
                "ForAllValues:StringEquals": {
                  "route53:ChangeResourceRecordSetsActions": [
                    "UPSERT",
                  ],
                  "route53:ChangeResourceRecordSetsRecordTypes": [
                    "CNAME",
                  ],
                },
                "ForAllValues:StringLike": {
                  "route53:ChangeResourceRecordSetsNormalizedRecordNames": [
                    "*.tekitou.ga",
                  ],
                },
              },
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":route53:::hostedzone/",
                    {
                      "Ref": "PublicHostedZone49973947",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CertificateManagerCertificateCertificateRequestorFunctionServiceRoleDefaultPolicy6F05AD3D",
        "Roles": [
          {
            "Ref": "CertificateManagerCertificateCertificateRequestorFunctionServiceRole39CBE3CA",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CertificateManagerCertificateCertificateRequestorResourceCEC8AEDD": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DomainName": "tekitou.ga",
        "HostedZoneId": {
          "Ref": "PublicHostedZone49973947",
        },
        "Region": "us-east-1",
        "ServiceToken": {
          "Fn::GetAtt": [
            "CertificateManagerCertificateCertificateRequestorFunction9E036FFB",
            "Arn",
          ],
        },
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": {
      "DependsOn": [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "6ddcf10002539818a9256eff3fb2b22aa09298d8f946e26ba121c175a600c44e.zip",
        },
        "Handler": "index.handler",
        "Layers": [
          {
            "Ref": "DeployWebsiteAwsCliLayer17DBC421",
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "s3bucket64CB25AF",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "s3bucket64CB25AF",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": [
          {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "3f51abb709b8e65167a45aeed02bab11540603d909005d7148230ba5ce6c74d7.zip",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "s3bucket64CB25AF",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DeployWebsiteAwsCliLayer17DBC421": {
      "Properties": {
        "Content": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "c409e6c5845f1f349df8cd84e160bf6f1c35d2b060b63e1f032f9bd39d4542cc.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "DeployWebsiteCustomResourceD116527B": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DestinationBucketName": {
          "Ref": "s3bucket64CB25AF",
        },
        "Prune": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": [
          {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
        ],
        "SourceObjectKeys": [
          "8314dad3d022fdb83a5674fe7fed2252293d7d510a30730ed16f50171cde5591.zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "MyOriginAccessIdentity96ED0331": {
      "Properties": {
        "CloudFrontOriginAccessIdentityConfig": {
          "Comment": "Unique Domain Hosting Environment",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "PublicHostedZone49973947": {
      "Properties": {
        "Name": "tekitou.ga.",
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "myDist9DB766F3": {
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "tekitou.ga",
          ],
          "DefaultCacheBehavior": {
            "AllowedMethods": [
              "GET",
              "HEAD",
            ],
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "CachedMethods": [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "TargetOriginId": "MyTestStackmyDistOrigin13B2398F7",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2and3",
          "IPV6Enabled": true,
          "Origins": [
            {
              "CustomOriginConfig": {
                "OriginProtocolPolicy": "http-only",
                "OriginSSLProtocols": [
                  "TLSv1.2",
                ],
              },
              "DomainName": {
                "Fn::Select": [
                  2,
                  {
                    "Fn::Split": [
                      "/",
                      {
                        "Fn::GetAtt": [
                          "s3bucket64CB25AF",
                          "WebsiteURL",
                        ],
                      },
                    ],
                  },
                ],
              },
              "Id": "MyTestStackmyDistOrigin13B2398F7",
            },
          ],
          "PriceClass": "PriceClass_All",
          "ViewerCertificate": {
            "AcmCertificateArn": {
              "Fn::GetAtt": [
                "CertificateManagerCertificateCertificateRequestorResourceCEC8AEDD",
                "Arn",
              ],
            },
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "myportfoliotest1BB247FD": {
      "Properties": {
        "Name": "my-portfolio-test",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "myportfoliotestAccount254DC821": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "myportfoliotest1BB247FD",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "myportfoliotestCloudWatchRole38D0CD69",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "myportfoliotestCloudWatchRole38D0CD69": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "myportfoliotestDeployment50C16562ecfb83d8bb1e13f72fa58229d482059b": {
      "DependsOn": [
        "myportfoliotestOPTIONSD4C37042",
        "myportfoliotestPOSTF316C6C7",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "myportfoliotest1BB247FD",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "myportfoliotestDeploymentStageprodA4566A1E": {
      "DependsOn": [
        "myportfoliotestAccount254DC821",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "myportfoliotestDeployment50C16562ecfb83d8bb1e13f72fa58229d482059b",
        },
        "RestApiId": {
          "Ref": "myportfoliotest1BB247FD",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "myportfoliotestOPTIONSD4C37042": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": {
          "IntegrationResponses": [
            {
              "ResponseParameters": {
                "method.response.header.Access-Control-Allow-Headers": "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'*'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": [
          {
            "ResponseParameters": {
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": {
          "Fn::GetAtt": [
            "myportfoliotest1BB247FD",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "myportfoliotest1BB247FD",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "myportfoliotestPOSTF316C6C7": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": {
          "Type": "MOCK",
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "myportfoliotest1BB247FD",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "myportfoliotest1BB247FD",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "s3bucket64CB25AF": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "my-port-folio-yh-tekitou-cdk",
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "IgnorePublicAcls": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          {
            "Key": "aws-cdk:cr-owned:0bbd7f95",
            "Value": "true",
          },
        ],
        "WebsiteConfiguration": {
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "s3bucketAutoDeleteObjectsCustomResource839C6F53": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "s3bucketPolicyF7E91061",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "s3bucket64CB25AF",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "s3bucketPolicyF7E91061": {
      "Properties": {
        "Bucket": {
          "Ref": "s3bucket64CB25AF",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "s3bucket64CB25AF",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "s3bucket64CB25AF",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "s3bucket64CB25AF",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "MyOriginAccessIdentity96ED0331",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "s3bucket64CB25AF",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
